File: src/com/meteorcode/pathway/model/Entity.java
package com.meteorcode.pathway.model;

public class Entity extends GameObject {
	
	/**
	 * Constructor for a new Entity. The GameID will be assigned by the
	 * GameEngine (NYI)
	 */
	public Entity() {
		super();
	}

	/**
	 * Constructor for an Entity with a known gameID. Calls the known-gameID
	 * constructor in GameObject.
	 * 
	 * @param gameID
	 *            the gameID for this Entity
	 */
	public Entity(long gameID) {
		super(gameID);
	}
}

File: src/com/meteorcode/pathway/model/Entity.scala
package com.meteorcode.pathway.model

import GridCoordinatesImplicits._

class Entity(gameID: Long,
             protected var coordinates: GridCoordinates,
             protected var grid: Grid,
             protected var name: String) extends GameObject(gameID) {

  if (this.name == null) this.name = this.getClass.getSimpleName

  def this() = this(null, null, null, null)
  def this(gameID: Long) = this (gameID, null, null, null)
  def this(position: (Int, Int), grid: Grid) = this (null, position, grid, null)
  def this(name: String) = this(null, null, null, name)
  def this(gameID: Long, name:String) = this (gameID, null, null, name)
  def this(position: (Int, Int), grid: Grid, name: String) = this (null, position, location, name)

  def setLocation (xy: GridCoordinates) = this.coordinates = (xy.x, xy.y)
  def getCoordinates: GridCoordinates = new GridCoordinates(coordinates)
  def getLocation: Tile = grid.getTileAt(coordinates)

}